mlir/lib/Dialect/Linalg/Transforms/Tiling.cpp

```diff
@@ -107,6 +107,138 @@ static void emitIsPositiveIndexAssertion(ImplicitLocOpBuilder &b,
       b.getStringAttr("expected strictly positive tile size and divisor"));
 }
 
+FailureOr<StaticContinuousTileSizeSpecification>
+mlir::linalg::computeStaticContinuousTileSizes(TilingInterface op,
+                                               unsigned dimension,
+                                               unsigned targetSize) {
+
+  LinalgOp linalgOp = dyn_cast<LinalgOp>(op.getOperation());
```

Comment:
There's no check that `linalgOp` is non-null here. Given that the code below assumes `op` is a Linalg op, I'd just take a `LinalgOp op` in this function and let the caller do the casting.

Reply:
Fixed. Reverted changes here to use LinalgOp instead of TilingInterface.